test/integration/new-storage/BucketManager.test.js

```diff
@@ -0,0 +1,142 @@
+const cassandra = require('cassandra-driver')
+const { waitForCondition, wait } = require('streamr-test-utils')
+
+const BucketManager = require('../../../src/new-storage/BucketManager')
+
+const contactPoints = ['127.0.0.1']
+const localDataCenter = 'datacenter1'
+const keyspace = 'streamr_dev'
+
+describe('BucketManager', () => {
+    let bucketManager
+    let streamId
+    let cassandraClient
+    let streamIdx = 1
+
+    beforeEach(async () => {
+        cassandraClient = new cassandra.Client({
+            contactPoints,
+            localDataCenter,
+            keyspace,
+        })
+
+        await cassandraClient.connect()
+        bucketManager = new BucketManager(cassandraClient, {
+            logErrors: true,
+            checkFullBucketsTimeout: 1000,
+            storeBucketsTimeout: 1000,
+            maxBucketSize: 10 * 300,
+            maxBucketRecords: 10,
+            bucketKeepAliveSeconds: 60
+        })
+
+        streamId = `stream-id-${Date.now()}-${streamIdx}`
+        streamIdx += 1
+    })
+
+    afterEach(async () => {
+        bucketManager.stop()
+        await cassandraClient.shutdown()
+    })
+
+    test('calling getBucketId() will try to find bucket in database and then create if not found and store in database', async () => {
+        const timestamp = Date.now()
+
+        const storeBucketsSpy = jest.spyOn(bucketManager, '_storeBuckets')
+
+        expect(Object.values(bucketManager.streams)).toHaveLength(0)
+        expect(Object.values(bucketManager.buckets)).toHaveLength(0)
+
+        expect(bucketManager.getBucketId(streamId, 0, timestamp)).toBeUndefined()
+        let result = await cassandraClient.execute('SELECT * FROM bucket WHERE stream_id = ? ALLOW FILTERING', [
+            streamId
+        ])
+        expect(result.rows.length).toEqual(0)
+
+        // first time we call in constructor, second after timeout
+        await waitForCondition(() => storeBucketsSpy.mock.calls.length === 2)
+        expect(storeBucketsSpy).toHaveBeenCalled()
```

Comment:
It's needed for assertion. `waitForCondition` is used only for waiting.

Reply:
I mean if `waitForCondition(() => storeBucketsSpy.mock.calls.length === 2)` fails.. well the test fails. If `waitForCondition(() => storeBucketsSpy.mock.calls.length === 2)` passes, well you know storeBucketsSpy has been called because the condition passed?

Reply:
Anyway just a small details so I'll resolve.